ci/infra/testrunner/tests/conftest.py

```diff
@@ -45,11 +45,11 @@ def bootstrap(request, provision, skuba):
 
 
 @pytest.fixture
-def deployment(request, bootstrap, skuba):
-    if request.config.getoption("skip_setup") == 'deployed':
-        return
+def deployment(request, bootstrap, skuba, kubectl):
+    if request.config.getoption("skip_setup") != 'deployed':
+        skuba.join_nodes()
 
-    skuba.join_nodes()
+    kubectl.run_kubectl('wait --timeout=5m --for=condition=Ready pods --all --namespace=kube-system')
```

Comment:
According to my tests the timeout applies for each pod, not to all tests. Therefore 5m for each pod seems excessive

with timeout=5s
```
time kubectl --kubeconfig=admin.conf wait --timeout=5s --for=condition=Ready pods --all --namespace=kube-system 
pod/cilium-lh9k5 condition met
pod/cilium-n78xb condition met
pod/cilium-operator-7d6ddddbf5-92tbq condition met
pod/cilium-xlr88 condition met
pod/etcd-master-0 condition met
pod/kube-apiserver-master-0 condition met
pod/kube-controller-manager-master-0 condition met
pod/kube-proxy-bvfhq condition met
pod/kube-proxy-cr8q4 condition met
pod/kube-proxy-rctzs condition met
pod/kube-scheduler-master-0 condition met
pod/kured-kwkdk condition met
pod/kured-tpvfx condition met
pod/oidc-dex-55fc689dc-vtssq condition met
pod/oidc-gangway-7b7fbbdbdf-j67q8 condition met
timed out waiting for the condition on pods/coredns-69c4947958-jjvcc
timed out waiting for the condition on pods/coredns-69c4947958-lcp8d
timed out waiting for the condition on pods/kured-9xsnb

real	0m19.337s
user	0m0.199s
sys	0m0.005s
```
with timeout=10s
```
real	0m36.136s
user	0m0.188s
sys	0m0.036s
```

Reply:
Ok I'll try a lower timeout

Reply:
Tried a few different timeout 10s, 1m, 2m, 3m. 3m seemed to work best